src/sentry/replays/query.py

```diff
@@ -599,6 +615,7 @@ def _activity_score():
         "min", parameters=[Column("replay_start_timestamp")], alias="started_at"
     ),
     "finished_at": Function("max", parameters=[Column("timestamp")], alias="finished_at"),
+    # Because duration considers the max timestamp, archive requests can alter the duration.
     "duration": Function(
```

Comment:
i think this will end up being a problem in other flows where we append non-session data to the replay. could the duration only look at rows for which segment_id is not null?

Reply:
We could aggregate the (timestamp, segment_id) into a tuple and then filter out where segment_id is null.

Reply:
👍🏼 yeah worth doing at some point, priority low right now though i think (backend errors e.g. should have a timestamp within the range of the replays timestamp)